user_management/tests/nh_clinical_user_management/test_write.py

```diff
@@ -1,65 +1,89 @@
     """
-
-    def setUp(self):
-        super(TestUserCRUD, self).setUp()
+    def call_test(self, user=None):
         hca_data = {
             'name': 'HCA2',
             'login': 'hca2',
             'category_id': [[4, self.hca_role.id]]
         }
-        self.hca = self.user_management_model\
-            .sudo(self.shift_coordinator).create(hca_data)
+        if user:
+            self.user_management_model = self.user_management_model.sudo(user)
+        self.hca = self.user_management_model.create(hca_data)
 
-    def test_creates_user(self):
-        """
-        Test that creating a user, creates a user
-        """
-        self.assertTrue(self.hca, msg="HCA user not created")
-        self.assertEqual(self.hca.name, 'HCA2')
-        self.assertEqual(self.hca.login, 'hca2')
+    def _assert_subordinate_updated(self, update):
+        self.assertTrue(update)
+        self.assertEqual(self.hca.name, 'TU2')
+        self.assertEqual(self.hca.login, 'tu2')
         groups = [g.name for g in self.hca.groups_id]
-        self.assertTrue('NH Clinical HCA Group' in groups,
-                        msg="User created without HCA group")
-        self.assertTrue('Employee' in groups,
-                        msg="User created without Employee group")
+        self.assertFalse('NH Clinical HCA Group' in groups,
+                         msg="HCA group not removed")
+        self.assertTrue('NH Clinical Nurse Group' in groups,
+                        msg="Nurse group missing")
+        self.assertTrue('Employee' in groups, msg="Employee group removed")
 
     def test_can_update_subordinate(self):
         """
-        Test that superior users can update subordinate user groups
+        Test that superior users can update subordinate user groups.
         """
+        self.call_test(user=self.shift_coordinator)
         update_data = {
             'name': 'TU2',
             'login': 'tu2',
             'category_id': [[6, 0, [self.nurse_role.id]]]
         }
         update = self.hca.sudo(self.shift_coordinator).write(update_data)
-        self.assertTrue(update)
-        self.assertEqual(self.hca.name, 'TU2')
-        self.assertEqual(self.hca.login, 'tu2')
-        groups = [g.name for g in self.hca.groups_id]
-        self.assertFalse('NH Clinical HCA Group' in groups,
-                         msg="HCA group not removed")
-        self.assertTrue('NH Clinical Nurse Group' in groups,
-                        msg="Nurse group missing")
-        self.assertTrue('Employee' in groups, msg="Employee group removed")
+        self._assert_subordinate_updated(update)
+
+    def test_superuser_can_update_subordinate(self):
+        """
+        Test that the superuser can update subordinate user groups.
+        """
+        self.call_test()
+        update_data = {
+            'name': 'TU2',
+            'login': 'tu2',
+            'category_id': [[6, 0, [self.nurse_role.id]]]
+        }
+        update = self.hca.write(update_data)
```

Comment:
Could we make it so it updated it to be a 'System Admin'? That way we cover for the 1 clear need we have for the superuser setting roles :)